### PR TITLE
Run `x test --stage 1` in CI

### LINKED ISSUE #99135: Run `x test --stage 1` in CI
Similar to https://github.com/rust-lang/rust/issues/95996, we expect contributors to run this frequently, so it should be tested in CI.

@eddyb suggested using the defaults in https://github.com/rust-lang/rust/blob/7443cc2bd66a79aa5a624e11b9b0172e56c2da48/src/bootstrap/defaults/config.compiler.toml#L1 but I'm a little hesitant to do that because some of them are *very* wrong for CI (e.g. `download-ci-llvm`) and others shouldn't affect tests but make the compiler slower (e.g. `debug-logging = true`). I do agree we should enable incremental for these tests, though.

Not sure which builder to add these to - adding it to mingw-check would make it significantly slower, and I don't know which of the full builders has spare capacity. @pietroalbini do you have suggestions?

_Originally posted by @eddyb in https://github.com/rust-lang/rust/pull/98660#issue-1288306531

### PR DESCRIPTION
**Pull request by Milo123459**
Fixes #99135

r? @jyn514

### CODE REVIEW

File: src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile
Comment (jyn514) [inline src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile:53]: you can skip these, they're already run in the other builder

File: src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile
Comment (jyn514) [inline src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile:44]: I suspect this will fail unless you add a bunch of excludes, see the list in https://github.com/rust-lang/rust/pull/99141/files#diff-5af50d95255aad85107312975f70a64e701c1eeee029cc933205297fd3863992R38-R40
but maybe those are ok with --stage 1 and only break with --stage 0? not sure, fine to leave as is unless CI fails

**Review by jyn514 — CHANGES_REQUESTED** [review]

File: src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile
Comment (Milo123459) [reply src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile:44]: yeah, will see if ci fails.

Comment (Milo123459) [reply src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile]: ci isn't failing, will resolve

**Comment by jyn514** [comment]
I think this will *only* run on PRs and not the main bors merge? that seems not great, it means you can hit semantic conflicts (although I guess those are rare in practice).

@pietroalbini how can I tell if this will run the new tests on a full merge or not?

**Comment by jyn514** [comment]
err pietro is out - will just ask someone random I guess.

r? rust-lang/infra-ci

**Comment by jyn514** [comment]
highfive pls

r? rust-lang/infra-ci

**Comment by Mark-Simulacrum** [comment]
The new builder was only added to the pr: stanza in the configuration, so it won't run on a full build.

If we're going to run the builder in PR CI, I would probably recommend we drop fulldeps tests and the other bits that are still causing two full compiler builds for it -- IIRC, @jyn514 you were looking into that on Zulip at some point for other reasons, though I forget the details.

**Comment by jyn514** [comment]
yeah, that was https://github.com/rust-lang/rust/pull/99141 - it ended up not having any time savings at all, which I think means stage 2 was still being compiled. Haven't had time to investigate - not sure it makes sense to block this change because of that.

**Comment by jyn514** [comment]
anyway, @Milo123459 I'm happy to approve if you change this to run in the full test suite as well as in PR checks

**Comment by Mark-Simulacrum** [comment]
I'm not sure it makes sense to run this in PR checks. There's not really a time benefit that I can see, and those builders will add up pretty quickly given how many PRs we have. Hopefully, defects here are pretty rare, so it doesn't seem like a good trade-off to me to spend that CI budget on this builder.

Is there some evidence that breaking stage 1 is common enough that we want the early warning at PR time rather than bors time?

@bors r-

**Comment by jyn514** [comment]
ah, fair enough - I still had it in my mind that we were doing this instead of stage2 builds, which isn't the case. Yes, I agree it's better not to run in PR checks.

**Comment by Milo123459** [comment]
Alright, stage1 tests are no longer run in PRs, only in the full test-suite.

**Pull request merged**
